src/ctelnet.cpp

```diff
@@ -3337,6 +3349,9 @@ void cTelnet::postData()
     if (mpHost->mpConsole) {
         mpHost->mpConsole->printOnDisplay(mMudData, true);
     }
+    if (mpHost->mmcpServer /*&& !mpHost->mIsRemoteEchoingActive*/) {
```

Comment:
Are you sure about that bit you have commented out? When the MUD Server is doing the echoing (i.e. `mpHost->mIsRemoteEchoingActive` is `true`) it prevents the text that the user enters from being painted up on the main console (this is following the convention as described in **[RFC 875](https://www.rfc-editor.org/rfc/rfc857.txt)** that both ends do not echo in a particular direction at the same time) - this is typically under the situation where the user is typing in a password or other stuff that they probably would not want to be sent to anyone else...